### ISSUE TITLE
Fix S3400 FP: Don't report function with conditional compilation body

### ISSUE THREAD

**Issue by tkouba** [SonarSource/sonar-dotnet#9804]
### Description

On multi-platform or multi-target code, sometimes need to return constant on some platforms/targets. 

### Reproducer

```csharp
        public static bool IsGooglePlayServicesAvailable()
        {
#if ANDROID
            try
            {
                var context = Android.App.Application.Context;
                var result = Android.Gms.Common.GoogleApiAvailability.Instance
                    .IsGooglePlayServicesAvailable(context);

                return result == Android.Gms.Common.ConnectionResult.Success;
            }
            catch
            {
                return false;
            }
#else
            // iOS / Windows platform fallback 
            return false;
#endif
        }
```

### Product and Version

VisualStudio: 17.14.29, SonarQube ide 9.9.0.16495

**Comment by pavel-mikula-sonarsource** [comment]
Aho Tomáši,

Thank you for reporting this case, I confirm it as a False Positive.
